test/test_autograd.py

```diff
@@ -4980,6 +4980,17 @@ def assert_only_first_requires_grad(res):
                                          return_counts=return_counts)
                     assert_only_first_requires_grad(res)
 
+    def test_linalg_qr_r(self):
+        # torch.linalg.qr(mode='r') returns only 'r' and discards 'q', but
+        # without 'q' you cannot compute the backward pass. Check that
+        # linalg_qr_backward complains cleanly in that case.
+        inp = torch.randn((5, 7), requires_grad=True)
+        q, r = torch.linalg.qr(inp, mode='r')
+        assert q.shape == (0,)  # empty tensor
```

Comment:
Instead of a naked assert maybe assertEqual here?

Reply:
fixed by commit 9299f87186, together with a bunch of other naked asserts